osu.Game/Screens/OnlinePlay/Multiplayer/MultiplayerMatchSubScreen.cs

```diff
@@ -685,6 +687,19 @@ private void showUserStyleSelect()
             this.Push(new MultiplayerMatchFreestyleSelect(room, new PlaylistItem(item)));
         }
 
+        /// <summary>
+        /// Shows the results screen for a playlist item.
+        /// </summary>
+        private void showResults(PlaylistItem item)
+        {
+            if (!this.IsCurrentScreen() || client.Room == null || client.LocalUser == null)
+                return;
+
+            // fallback is to allow this class to operate when there is no parent OnlineScreen (testing purposes).
+            var targetScreen = (Screen?)parentScreen ?? this;
+            targetScreen.Push(new PlaylistItemUserBestResultsScreen(client.Room.RoomID, item, client.LocalUser.UserID));
```

Comment:
Here's a curveball: For head to head this *is* going to work fine... but what about team versus? I think the expectation would be that if the playlist item was played in team versus, then team totals should be displayed on the results screen. `PlaylistItemUserBestResultsScreen` is obviously incapable of doing that.

The second layer to this is that even if you agree with the above premise, it's not really even *possible* to determine whether the results should show a "head to head" results screen, or a "team vs" results screen, because room mode is only persisted against the room, and only the latest state of it. Room modes at the time of playing a playlist item are not persisted anywhere. I ran into this myself working on https://github.com/ppy/osu-web/issues/10455.

Reply:
After talking this through via other communication avenues I'm probably just going to look away for now until someone inevitably notices and complains.